kolibri/plugins/learn/assets/src/views/content-card-carousel/index.vue

```diff
@@ -0,0 +1,302 @@
+        const gutters = (this.contentSetSize - 1) * gutterWidth;
+        const carouselContainerOffset = cards + gutters;
+        const sign = this.leftToRight ? -1 : 1;
+        el.style.left = `${(sign * carouselContainerOffset) + originalPosition}px`;
+      },
+      slide(el) {
+        const originalPosition = parseInt(el.style.left, 10);
+        const cards = this.contentSetSize * contentCardWidth;
+        const gutters = (this.contentSetSize - 1) * gutterWidth;
+        const carouselContainerOffset = cards + gutters;
+        const sign = this.leftToRight ? 1 : -1;
+        el.style.left = `${(sign * carouselContainerOffset) + originalPosition}px`;
+      },
+      isInThisSet(index) {
+        return this.contentSetStart <= index && index <= this.contentSetEnd;
+      },
+      nextSet() {
+        this.contentSetStart += this.contentSetSize;
+        this.leftToRight = false;
+      },
+      previousSet() {
+        this.contentSetStart -= this.contentSetSize;
+        this.leftToRight = true;
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  // width of card + gutter
+  $card-height = 210px
+
+  .content-carousel
+    margin-top: 1em
+    margin-bottom: 1em
+    clearfix()
+
+    &-details
+      clearfix()
+
+      &-header
+        float: left
+        text-align: left
+        margin-bottom: 1em
+        h2
+          margin: 0
+      &-view-all
+        float: right
+        color: white
+        background-color: $core-action-normal
+
+
+    &-controls
+      $hit-height = 100px
+
+      $hit-width = $hit-height
+      // set up the parent element that the buttons use for reference
+      position: absolute
+      width: 100%
+      clearfix()
+
+      // styles that apply to both control buttons
+      .next, .previous
+        &:active
+          z-index: 8 // material
+          // goes up one reference (Stylus partial reference)
+
+        z-index: 2 // material
+        position: absolute
+        top: ($card-height / 2)
+        transform: translateY(-($hit-height / 2))
+        height: $hit-height
+        width: $hit-width
+        text-align: center
+        vertical-align: middle
+
+        &-button
+          &:active
+            box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23) // material
+          box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23)
+          // center align within hitbox
+          position: absolute
+          top: 50%
+          left: 50%
+          transform: translate(-50%, -50%)
+
+      // position-specific styles for each control button
+      .next
+        right: -($hit-width/2)
+      .previous
+        left: -($hit-width/2)
+
+    &-set
+      $max-card-shadow-offset = 10px
+      position: relative
+      height: $card-height + $max-card-shadow-offset
+      overflow-x: hidden
+      overflow-y: visible
+
+    &-card
+      transition: left 0.4s linear
```

Comment:
@indirectlylit which actually wouldn't be that hard given that the JS hooks in `transition-group` give us an element with styles in-lined